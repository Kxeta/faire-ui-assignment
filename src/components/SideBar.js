import React from 'react';
import { PropTypes } from 'prop-types';
import { Link, withRouter } from 'react-router-dom';
import { defaultCategory } from '../constants/utils';

import './Sidebar.css';

const Sidebar = props => {
  const isActualCategory = name => {
    const { actualCategory } = props.match.params;
    if (actualCategory && actualCategory === name) {
      return true;
    } else if (!actualCategory && name === defaultCategory) {
      return true;
    }
    return false;
  };
  return (
    <div className="sidebar">
      <ul>
        {props.list &&
          props.list.map((route, index) => (
            <li
              key={index}
              className={isActualCategory(route.name) ? 'actual-category' : ''}
            >
              {isActualCategory(route.name) ? (
                route.name
              ) : (
                <Link to={route.linkTo}>{route.name}</Link>
              )}
            </li>
          ))}
      </ul>
    </div>
  );
};

Sidebar.propTypes = {
  list: PropTypes.array,
};

export default withRouter(Sidebar);
