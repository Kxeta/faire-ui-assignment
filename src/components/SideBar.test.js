import React from 'react';
import { Sidebar } from './SideBar';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';

import '../setupTests.js';

describe('Rendering - Sidebar', () => {
  it('Should render without crashing', () => {
    const wrapper = shallow(<Sidebar />);
    expect(wrapper).toMatchSnapshot();
  });
  it('Sould render with props correctly', () => {
    const wrapper = shallow(
      <Sidebar list={[{ name: 'Champs', linkTo: '/champs' }]} />,
    );
    expect(wrapper.find(Link).length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
