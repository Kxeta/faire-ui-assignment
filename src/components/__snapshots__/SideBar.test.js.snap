// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering - Sidebar Should render without crashing 1`] = `
<div
  className="sidebar"
>
  <ul />
</div>
`;

exports[`Rendering - Sidebar Sould render with props correctly 1`] = `
<div
  className="sidebar"
>
  <ul>
    <li
      className=""
      key="0"
    >
      <Link
        to="/champs"
      >
        Champs
      </Link>
    </li>
  </ul>
</div>
`;
